src/Native/MatrixFactorizationNative/UnmanagedMemory.h

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#pragma once
+#include "mf.h"
+
+using namespace mf;
+
+#ifdef _MSC_VER
+#define EXPORT_API(ret) extern "C" __declspec(dllexport) ret __stdcall
+#else
+#define EXPORT_API(ret) extern "C" __attribute__((visibility("default"))) ret
+#endif
```

Comment:
we have a [https://github.com/dotnet/machinelearning/blob/master/src/Native/Stdafx.h](https://github.com/dotnet/machinelearning/blob/master/src/Native/Stdafx.h).
I would suggest to link it, and don't define them again.

Reply:
Done.